m3u8/writer.go

```diff
@@ -1069,3 +1334,63 @@ func (p *MediaPlaylist) GetAllSegments() []*MediaSegment {
 /*
 [Protocol Version Compatibility]: https://datatracker.ietf.org/doc/html/draft-pantos-hls-rfc8216bis-16#section-8
 */
+
+/// Helper functions
+
+func splitUriBy(uri, sep string) (string, string) {
+	// split the uri by the last dot
+	uriParts := strings.Split(uri, sep)
+	if len(uriParts) < 2 {
+		return "", ""
+	}
+	// get the last part of the uri
+	lastPart := uriParts[len(uriParts)-1]
+	// get the rest of the uri
+	rest := strings.Join(uriParts[:len(uriParts)-1], sep)
+	return rest, lastPart
+}
+
+// find the numbers in end of the string
+func getSequenceNum(uriPrefix string) (bool, uint64) {
```

Comment:
Follow the doc string pattern and write this as `// getSequenceNum return  the last number in uriPrefix`.
Normally, the bool value is the second parameter, for example when checking a value in a map.

I'd recommend to use named return values here `(num uint64, ok bool)` to make the function signature clearer.